modules/candidate_parameters/jsx/monitoring.js

```diff
@@ -0,0 +1,449 @@
+
+          visits.push(
+            <SelectElement
+              label ={label}
+              name ={visit}
+              options ={this.state.flaggedOptions}
+              value ={this.state.formData[visit]}
+              onUserInput ={this.setFormData}
+              ref ={visit}
+              disabled ={disabled}
+              required ={true}
+            />
+          );
+          visits.push(
+            <SelectElement
+              label ={monitoredLabel}
+              name ={monitored}
+              options ={this.state.monitoredOptions}
+              value ={this.state.formData[monitored]}
+              onUserInput ={this.setFormData}
+              ref ={monitored}
+              disabled ={disabledFields[i] || disabled}
+              required ={monitoredRequired[i]}
+            />
+          );
+          visits.push(
+            <DateElement
+              label ={dateMonitoredLabel}
+              name ={dateMonitored}
+              value ={this.state.formData[dateMonitored]}
+              onUserInput ={this.setFormData}
+              ref ={dateMonitored}
+              disabled ={disabledFields[i] || disabled}
+              required ={elRequired[i]}
+            />
+          );
+          visits.push(
+            <DateElement
+              label ={dateMonitored2Label}
+              name ={dateMonitored2}
+              value ={this.state.formData[dateMonitored2]}
+              onUserInput ={this.setFormData}
+              ref ={dateMonitored2}
+              disabled ={disabledFields[i] || disabled}
+              required ={elRequired[i]}
+            />
+          );
+          visits.push(
+            <SelectElement
+              label ={monitorIDLabel}
+              name ={monitorID}
+              options ={this.state.Data.examiners}
+              value ={this.state.formData[monitorID]}
+              onUserInput ={this.setFormData}
+              ref ={monitorID}
+              disabled ={disabledFields[i] || disabled}
+              required ={elRequired[i]}
+            />
+          );
+
+          visits.push(<hr />);
+          i++;
+        }
+      }
+
+      var formattedHistory = [];
+      for (let row in this.state.Data.history) {
+        if (this.state.Data.history.hasOwnProperty(row)) {
+          let line = [];
+          let lineInfo = "";
+          let historyRow = this.state.Data.history[row];
+          let userH = historyRow.entry_staff;
+          let updateDateH = historyRow.data_entry_date;
+          let visitH = historyRow.visit_label;
+          let flaggedH = historyRow.flag;
+          let monitoredH = historyRow.monitored;
+          let monitoredDateH = historyRow.date_monitored;
+          let examinerH = historyRow.monitor_id;
+
+          lineInfo += "[UPDATE by ";
+          lineInfo += userH;
+          lineInfo += " at ";
+          lineInfo += updateDateH;
+          lineInfo += "]";
+
+          line.push(
```

Comment:
You can define the whole line as a block of JSX instead of pushing for no reason